File: AtCoder/ABC169/b.cpp
#include <bits/stdc++.h>

using namespace std;
typedef unsigned long long ull;
int main(){
    int n;
    ull MAX = pow(10,18);
    cin >> n;
    ull res = 1;
    for (int i=0;i<n;i++){
        ull num = 0;
        cin >> num;
        res*=num;

    }
    if (res>MAX){
    cout << "-1";
    }
    else
        cout << res<<endl;
}

File: AtCoder/ABC169/b.py
#TIME : 
import sys
from math import sqrt
input = sys.stdin.readline

def inp():
    return(int(input()))
def inlt():
    return(list(map(int,input().split())))
def insr():
    s = input()
    return(list(s[:len(s) - 1]))
def invr():
    return(map(int,input().split()))

MAX = 10**18
n = inp()
a = inlt()
if 0 in a:
    print(0)
    exit()
res = 1
for num in a:
    res*=num
    if res > MAX:
        print(-1)
        exit()
print(res)